sources/functions/requestrr

```diff
@@ -0,0 +1,166 @@
+        "armhf") dlurl=$(curl -sNL $urlbase | grep -Po 'ht(.*)linux-arm(.*)zip') >> ${log} 2>&1 ;;
+        "arm64") dlurl=$(curl -sNL $urlbase | grep -Po 'ht(.*)linux-arm64(.*)zip') >> ${log} 2>&1 ;;
+        *)
+            echo_error "Arch not supported"
+            exit 1
+            ;;
+    esac
+
+    if ! curl "$dlurl" -L -o /tmp/requestrr.zip >> "$log" 2>&1; then
+        echo_error "Download failed, exiting"
+        exit 1
+    fi
+    echo_progress_done "Source downloaded"
+}
+
+function _requestrr_chpass() {
+    password="$(_get_user_password "$(_get_master_username)")"
+    local_packages=/usr/local/bin/swizzin
+    readarray -t hashpass < <(python3 ${local_packages}/requestrr.hashPass.py "${password}")
+    export requestrr_pass=${hashpass[0]}
+    export requestrr_key=${hashpass[1]}
+}
+
+function _requestrr_config() {
+    _requestrr_chpass
+    if [[ -f /install/.sonarr.lock ]]; then
+        s_config="/home/$user/.config/sonarr/config.xml"
+        s_bind=$(sed -n 's|\(.*\)<BindAddress>\(.*\)</BindAddress>|\2|p' "$s_config")
+        s_port=$(sed -n 's|\(.*\)<Port>\(.*\)</Port>|\2|p' "$s_config")
+        s_api=$(sed -n 's|\(.*\)<ApiKey>\(.*\)</ApiKey>|\2|p' "$s_config")
+        s_base=$(sed -n 's|\(.*\)<UrlBase>\(.*\)</UrlBase>|\2|p' "$s_config")
+        sonarr=("$s_bind" "$s_port" "$s_api" "$s_base")
+    fi
+    if [[ -f /install/.radarr.lock ]]; then
+        r_config="/home/$user/.config/Radarr/config.xml"
+        r_bind=$(sed -n 's|\(.*\)<BindAddress>\(.*\)</BindAddress>|\2|p' "$r_config")
+        r_port=$(sed -n 's|\(.*\)<Port>\(.*\)</Port>|\2|p' "$r_config")
+        r_api=$(sed -n 's|\(.*\)<ApiKey>\(.*\)</ApiKey>|\2|p' "$r_config")
+        r_base=$(sed -n 's|\(.*\)<UrlBase>\(.*\)</UrlBase>|\2|p' "$r_config")
+        radarr=("$r_bind" "$r_port" "$r_api" "$r_base")
+    fi
+    cat > /opt/requestrr/SettingsTemplate.json << CFG
+{
+  "Authentication": {
+    "Username": "$(_get_master_username)",
+    "Password": "${requestrr_pass}",
+    "PrivateKey": "${requestrr_key}"
+  },
+  "ChatClients": {
+    "Discord": {
+      "BotToken": "",
+      "ClientId": "",
+      "StatusMessage": "!help",
+      "TvShowRoles": [],
+      "MovieRoles": [],
+      "MonitoredChannels": [],
+      "EnableRequestsThroughDirectMessages": false,
+      "AutomaticallyNotifyRequesters": true,
+      "NotificationMode": "PrivateMessages",
+      "NotificationChannels": [],
+      "AutomaticallyPurgeCommandMessages": false,
+      "DisplayHelpCommandInDMs": true
+    }
+  },
+  "DownloadClients": {
+    "Ombi": {
+      "Hostname": "",
+      "Port": 3579,
+      "ApiKey": "",
+      "ApiUsername": "",
+      "BaseUrl": "",
+      "UseSSL": false,
+      "Version": "3"
+    },
+    "Overseerr": {
+      "Hostname": "",
+      "Port": 5055,
+      "ApiKey": "",
+      "DefaultApiUserID": "",
+      "UseSSL": false,
+      "Version": "1"
+    },
+    "Radarr": {
+      "Hostname": "${radarr[0]}",
+      "Port": ${radarr[1]},
```

Comment:
This value needs to have a default, when radarr is not installed, otherwise the JSON is invalid.

Reply:
```json
    "Overseerr": {
      "Hostname": "",
      "Port": 5055,
      "ApiKey": "",
      "DefaultApiUserID": "",
      "UseSSL": false,
      "Version": "1"
    },
    "Radarr": {
      "Hostname": "",
      "Port": , // <<< This is what ya gets, and that ain't valid
      "ApiKey": "",
      "BaseUrl": "",
      "MovieProfileId": "1",
      "MovieRootFolder": "",
      "MovieMinimumAvailability": "",
      "MovieTags": [],
      "AnimeProfileId": "1",
```

Reply:
Ah. Forgot to set it as an else 8989 default.

Reply:
Could use this
```bash
FOO="${VARIABLE:-default}"  # If variable not set or null, use "default".
```